CMakeLists.txt

```diff
@@ -454,6 +454,40 @@ foreach(gen_src ${surelog_generated_SRC})
   set_source_files_properties(${gen_src} PROPERTIES GENERATED TRUE)
 endforeach()
 
+# Parse out version and configure header
+if(NOT DEFINED ENV{SURELOG_VERSION_MAJOR})
+  # parse from working directory
+  # ensure `git` is installed
+  find_package (Git REQUIRED)
+
+  # get latest commit
+  execute_process(COMMAND git describe --tags --abbrev=0
+    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+    OUTPUT_VARIABLE LATEST_VERSION)
+
+  # parse out first number after "v"
+  string(REGEX REPLACE "^v([0-9]+)\\..*" "\\1" SURELOG_VERSION_MAJOR "${LATEST_VERSION}")
+  # parse out second number, e.g. after "v[0-9]."
+  string(REGEX REPLACE "^v[0-9]+\\.([0-9]+).*" "\\1" SURELOG_VERSION_MINOR "${LATEST_VERSION}")
+
+  # get latest commit
+  execute_process(COMMAND git rev-parse HEAD
+    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+    OUTPUT_VARIABLE SURELOG_VERSION_COMMIT_SHA)
+  # strip newline
+  string(REGEX REPLACE "\n$" "" SURELOG_VERSION_COMMIT_SHA "${SURELOG_VERSION_COMMIT_SHA}")
+else()
+  # if set in env, assume some other deployment mechanism where
+  # not working from the git repo / where git not installed
+  set(SURELOG_VERSION_MAJOR ENV{SURELOG_VERSION_MAJOR})
```

Comment:
Similar to the UHDM suggestion, maybe we should have 
```
project(surelog VERSION 1.48)
```

... and then derive values from there, so that it works out-of-the box for someone who has downloaded the source archive.